File: skills/lasersell/examples/build-sell/build_sell.py
"""Build an unsigned sell transaction with the LaserSell API.

Before running:
- Replace API key, mint, and wallet placeholders.
- Set `amount_tokens` in mint atomic units.
"""

from __future__ import annotations

import asyncio

from lasersell_sdk.exit_api import BuildSellTxRequest, ExitApiClient, SellOutput


async def main() -> None:
    client = ExitApiClient.with_api_key("REPLACE_WITH_API_KEY")

    request = BuildSellTxRequest(
        mint="REPLACE_WITH_MINT",
        user_pubkey="REPLACE_WITH_WALLET_PUBKEY",
        amount_tokens=1_000_000,
        slippage_bps=2_000,
        output=SellOutput.SOL,
    )

    response = await client.build_sell_tx(request)
    print(f"unsigned_tx_b64={response.tx}")
    if response.route is not None:
        print(f"route={response.route}")


if __name__ == "__main__":
    asyncio.run(main())

File: skills/lasersell/examples/build-sell/build_sell.ts
/**
 * Build an unsigned sell transaction with the LaserSell API.
 *
 * Before running:
 * - Replace API key, mint, and wallet placeholders.
 * - Set `amount_tokens` in mint atomic units.
 *
 * This example prints:
 * - `unsigned_tx_b64`
 * - optional `route` metadata
 */
import {
  ExitApiClient,
  type BuildSellTxRequest,
} from "@lasersell/lasersell-sdk";

async function main(): Promise<void> {
  const client = ExitApiClient.withApiKey("REPLACE_WITH_API_KEY");

  const request: BuildSellTxRequest = {
    mint: "REPLACE_WITH_MINT",
    user_pubkey: "REPLACE_WITH_WALLET_PUBKEY",
    amount_tokens: 1_000_000,
    slippage_bps: 2_000,
    output: "SOL",
  };

  const response = await client.buildSellTx(request);
  console.log(`unsigned_tx_b64=${response.tx}`);
  if (response.route !== undefined) {
    console.log(`route=${JSON.stringify(response.route)}`);
  }
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});
